anagram: Added 2 hints (#837)
exercism/rust: exercises/anagram/.meta/hints.md

# Hints

> The solution is case insensitive, that means `"WOrd"` is the same with `"word"` or `"woRd"`. It may help to take a peak in the [std library](https://doc.rust-lang.org/std/index.html) for functions that can convert between them.

Reviewer: Fixing some typos:

```suggestion
The solution is case insensitive, which means `"WOrd"` is the same as `"word"` or `"woRd"`. It may help to take a peek at the [std library](https://doc.rust-lang.org/std/index.html) for functions that can convert between them.
```

If you merge this, don't forget to regenerate README.md

> The solution is case insensitive, that means `"WOrd"` is the same with `"word"` or `"woRd"`. It may help to take a peak in the [std library](https://doc.rust-lang.org/std/index.html) for functions that can convert between them.

Reviewer: I think that if we include a link, it should be to a more useful page than the docs homepage. Suggest https://doc.rust-lang.org/std/primitive.str.html. I would also accept removing the link here.

Author: from this link, the solution is one search away. I thing its good experience searching for what you want. Maybe I should add "search for char" or something similar

> The solution cannot contain the word itself. Basically a word is always an anagram of it's self but for this exercise we don't care. This mean, given `"hello"` and the list `["hello", "olleh"]` the answear is `["olleh"]`

Reviewer: More typos, grammar, and style:

```suggestion
The solution cannot contain the input word. A word is always an anagram of itself, which means it is not an interesting result. Given `"hello"` and the list `["hello", "olleh"]` the answer is `["olleh"]`.
```
